### PR TITLE
admin: Histogram html use summary-buckets mode by default

### PR DESCRIPTION
**Pull request by jmarantz**
Commit Message: The detailed histograms may be too detailed for the default HTML view. Instead, respect the buckets-mode, and only use detailed mode if this was requested by the user.

Also, fix an accidental script-driven change (#27987) to a JS file that made it invalid JS, breaking active_html mode.
Additional Description:
Risk Level: low
Testing: added new test for histogram modes in HTML, and manually rendered in a server.
Docs Changes: n/a -- this change makes the system behave as documented, rather than ignoring the buckets-mode for HTML.
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: source/server/admin/stats_html_render.cc
Comment (ravenblackx) [inline source/server/admin/stats_html_render.cc:79]: Shouldn't this comment stay in the header? The header is usually where we go to find a description of what a function does.
Also it probably should be updated to reflect the updated behavior.

**Review by ravenblackx — CHANGES_REQUESTED** [review]

File: source/server/admin/stats_html_render.cc
Comment (jmarantz) [reply source/server/admin/stats_html_render.cc:79]: Good point about updating it to the changed functionality :) But this is a virtual function override to just render the histogram, and (e.g.) text and prometheus and json all have different approaches to this.

Comment (ravenblackx) [reply source/server/admin/stats_html_render.cc]: But this *specific* virtual function override is declared in `stats_html_render.h`, right? So there'll only be one implementation associated with that declaration. I'm not saying the comment belongs on the base class, but on the declaration for the function (where it was before).

Comment (jmarantz) [reply source/server/admin/stats_html_render.cc]: I don't feel that strongly about it, but I think as a reader I'd prefer to see it at the implementation; usually `override` headers do not have comments about how that implementation differs, unless the impl is inline. I decided not to inline this one since it's got a conditional in it and it feels more aesthetic to me in the .cc file.

I don't think the Envoy style guide is specific about this, is it?

**Review by ravenblackx — DISMISSED** [review]

**Comment by jmarantz** [comment]
@ggreenway can you do a senior maintainer stamp?

**Review by ggreenway — CHANGES_REQUESTED** [review]
LGTM modulo one nit

/wait

**Review by ggreenway — APPROVED** [review]

**Comment by jmarantz** [comment]
Thanks Greg!

**Pull request merged**
